packages/cli/generators/relation/base-relation.generator.js

```diff
@@ -0,0 +1,204 @@
+    } else {
+      relationUtils.addProperty(classDeclaration, property);
+    }
+  }
+
+  _addParametersToRepositoryConstructor(classConstructor) {
+    const parameterName =
+      utils.camelCase(this.artifactInfo.dstRepositoryClassName) + 'Getter';
+
+    if (relationUtils.doesParameterExist(classConstructor, parameterName)) {
+      throw new Error(
+        'Parameter ' + parameterName + ' already exist in the constructor.',
+      );
+    }
+
+    classConstructor.addParameter({
+      decorators: [
+        {
+          name: 'repository.getter',
+          arguments: ["'" + this.artifactInfo.dstRepositoryClassName + "'"],
+        },
+      ],
+      name: parameterName,
+      type: 'Getter<' + this.artifactInfo.dstRepositoryClassName + '>,',
+      scope: ast.Scope.Protected,
+    });
+  }
+
+  _addCreatorToRepositoryConstructor(classConstructor) {
+    throw new Error('Not implemented');
+  }
+
+  _initializeProperties(options) {
+    // src configuration.
+    this.artifactInfo.srcModelPrimaryKey = options.sourceModelPrimaryKey;
+    this.artifactInfo.srcModelFile = path.resolve(
+      this.artifactInfo.modelDir,
+      utils.getModelFileName(options.sourceModel),
+    );
+
+    this.artifactInfo.srcModelClass = options.sourceModel;
+
+    this.artifactInfo.srcRepositoryFile = path.resolve(
+      this.artifactInfo.repositoryDir,
+      utils.getRepositoryFileName(options.sourceModel),
+    );
+
+    this.artifactInfo.srcRepositoryClassName =
+      utils.toClassName(options.sourceModel) + 'Repository';
+
+    this.artifactInfo.srcRepositoryFileObj = new relationUtils.AstLoopBackProject().addExistingSourceFile(
+      this.artifactInfo.srcRepositoryFile,
+    );
+
+    // dst configuration
+    this.artifactInfo.dstModelFile = path.resolve(
+      this.artifactInfo.modelDir,
+      utils.getModelFileName(options.destinationModel),
+    );
+
+    this.artifactInfo.dstModelClass = options.destinationModel;
+
+    this.artifactInfo.dstRepositoryFile = path.resolve(
+      this.artifactInfo.repositoryDir,
+      utils.getRepositoryFileName(options.destinationModel),
+    );
+
+    this.artifactInfo.dstRepositoryClassName =
+      utils.toClassName(options.destinationModel) + 'Repository';
+
+    // relation configuration
+    this.artifactInfo.relationName = options.relationName;
+  }
+
+  _getRepositoryRequiredImports(dstModelClassName, dstRepositoryClassName) {
+    return [
+      {
+        name: dstModelClassName,
+        module: '../models',
+      },
+      {
+        name: 'repository',
+        module: '@loopback/repository',
+      },
+      {
+        name: 'Getter',
+        module: '@loopback/core',
+      },
+      {
+        name: dstRepositoryClassName,
+        module: `./${utils.kebabCase(dstModelClassName)}.repository`,
+      },
+    ];
+  }
+
+  _getRepositoryRelationPropertyName() {
+    throw new Error('Not implemented');
+  }
+
+  _getRepositoryRelationPropertyType() {
+    throw new Error('Not implemented');
```

Comment:
Please add `/* istanbul ignore next */ ` before this line to tell the code coverage engine to ignore it.

```ts
/* istanbul ignore next */ 
throw new Error('Not implemented');
```

See https://github.com/gotwarlost/istanbul/blob/master/ignoring-code-for-coverage.md